### ISSUE TITLE
Failed to run on Arch Linux from 0.6.2

### ISSUE THREAD

**Issue by pikakolendo02** [Cygn/pulseaudio-dlna#38]
Hello Cygn,
I'm the maintainer of [this package](https://aur.archlinux.org/packages/pulseaudio-dlna-cygn) in AUR.
After I updated to 0.6.2 or later, I ran `pulseaudio-dlna` but got these error messages (below shows errors in 0.6.4):
```
/usr/lib/python3.10/site-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 2.0.5-build-libtorrent-rasterbar-src-libtorrent-rasterbar-2.0.5-bindings-python is an invalid version and will not be supported in a future release
  warnings.warn(
Traceback (most recent call last):
  File "/usr/local/bin/pulseaudio-dlna", line 33, in <module>
    sys.exit(load_entry_point('pulseaudio-dlna==0.6.3', 'console_scripts', 'pulseaudio-dlna')())
  File "/usr/local/bin/pulseaudio-dlna", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 162, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/lib/python3.10/site-packages/pulseaudio_dlna/__init__.py", line 19, in <module>
    import pkg_resources
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3266, in <module>
    def _initialize_master_working_set():
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3240, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3278, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 573, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 891, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 777, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'gobject' distribution was not found and is required by pulseaudio-dlna
```
These errors both appear from PKGBUILD ([0.6.4 ver](https://aur.archlinux.org/cgit/aur.git/commit/?h=pulseaudio-dlna-cygn&id=8437dc881b85c8f7c6b83ba1ed1e24d5a68c9a68)) and manually `make install`.

However, 0.6.1 runs without problem.

I'm not a native English speaker so please apologize my poor English expression.

**Comment by FabioLolix** [comment]
Hello

I can confirm that 'gobject' is not recognized with v0.6.4, while v0.6.1 start

@pikakolendo02 about the AUR pkgbuild, git is no longer needed as makedepends and please use the setup.py for building the package

```
build() {
  cd "${srcdir}/pulseaudio-dlna-${pkgver}" 
  python setup.py build
}

package() {
  cd "${srcdir}/pulseaudio-dlna-${pkgver}" 
  python setup.py install --root="$pkgdir" --optimize=1 --skip-build
}
```

**Comment by pikakolendo02** [comment]
Hello [FabioLolix](https://github.com/FabioLolix),
I removed git depend in PKGBUILD and used your scripts to build 0.6.4, however it returned the same errors.

```
pkgname=pulseaudio-dlna-cygn
pkgver=0.6.4
pkgrel=1
pkgdesc='Small DLNA server which brings DLNA/UPnP support to PulseAudio, Cygn edition'
arch=('x86_64')
url=https://github.com/Cygn/pulseaudio-dlna/
license=('GPL3')
provides=('pulseaudio-dlna')
depends=('python-chardet' 'python-dbus' 'python-docopt' 'python-gobject'
         'python-lxml' 'python-netifaces' 'python-notify2' 'python-psutil'
         'python-pychromecast' 'python-pyroute2' 'python-requests'
         'python-setproctitle' 'python-protobuf' 'python-setuptools'
         'python-zeroconf' 'python-urllib3' 'python-virtualenv')
makedepends=('help2man')
optdepends=('faac: AAC transcoding support'
            'ffmpeg: multiple formats support'
            'flac: FLAC transcoding support'
            'lame: MP3 transcoding support'
            'opus-tools: OPUS transcoding support'
            'sox: WAV transcoding support'
            'vorbis-tools: OGG transcoding support')
source=("${pkgname}-${pkgver}.tar.gz::https://github.com/Cygn/pulseaudio-dlna/archive/${pkgver}.tar.gz")
sha256sums=('SKIP')

# prepare() {
#   mv "pulseaudio-dlna-${pkgver}" "${pkgname}-${pkgver}"
#   cd "${pkgname}-${pkgver}"
#   sed -i '/dbus-python/d' setup.py
# }

build() {
  cd "${srcdir}/pulseaudio-dlna-${pkgver}" 
  python setup.py build
}

package() {
  cd "${srcdir}/pulseaudio-dlna-${pkgver}" 
  python setup.py install --root="$pkgdir" --optimize=1 --skip-build
}
```

**Comment by Cygn** [comment]
Hey folks,
Putting pygobject as requirement in setup.py somehow borks the rpmbuild process (remember that this repo is for fedora/copr/rpm building, not for dev) so I pull pyhton3-gobject as dep in the rpm.spec rather than in the setup.py. 
Surely something like that can be done in PKGBUILD ?  Or try to add pygobject in setup.py  install_requires list and see if it builds in Arch.

**Comment by pikakolendo02** [comment]
> Hey folks, Putting pygobject as requirement in setup.py somehow borks the rpmbuild process (remember that this repo is for fedora/copr/rpm building, not for dev) so I pull pyhton3-gobject as dep in the rpm.spec rather than in the setup.py. Surely something like that can be done in PKGBUILD ? Or try to add pygobject in setup.py install_requires list and see if it builds in Arch.

Hello Cygn,
`PyGObject` is already in `install_requires` in setup.py in 0.6.4.

**Comment by Cygn** [comment]
Did you try to build python3-rpm branch ? Or remove pygobject from setup requirements.

**Comment by FabioLolix** [comment]
v0.6.4.1 start

> Or remove pygobject from setup requirements.

pkgbuilds don't use or intereacts with requirements.txt

***

@pikakolendo02 seems that the pkgbuild could not use `python-virtualenv`, `help2man` and `make manpage` and simply installing the man pages with

`  install -D man/pulseaudio-dlna.1 -t "${pkgdir}/usr/share/man/man1/"`

arch=('x86_64') need to be changed to 'any'

from namcap

```
Checking pulseaudio-dlna-cygn-git-0.6.4.1.r0.gcbb87da-1-x86_64.pkg.tar.zst
pulseaudio-dlna-cygn-git W: No ELF files and not an "any" package
```

**Comment by pikakolendo02** [comment]
@FabioLolix Could you build an available package? I built 0.6.4.1 with this PKGBUILD
```
pkgname=pulseaudio-dlna-cygn
pkgver=0.6.4.1
pkgrel=1
pkgdesc='Small DLNA server which brings DLNA/UPnP support to PulseAudio, Cygn edition'
arch=('any')
url=https://github.com/Cygn/pulseaudio-dlna/
license=('GPL3')
provides=('pulseaudio-dlna')
depends=('python-chardet' 'python-dbus' 'python-docopt' 'python-gobject'
         'python-lxml' 'python-netifaces' 'python-notify2' 'python-psutil'
         'python-pychromecast' 'python-pyroute2' 'python-requests'
         'python-setproctitle' 'python-protobuf' 'python-setuptools'
         'python-zeroconf' 'python-urllib3')
makedepends=()
optdepends=('faac: AAC transcoding support'
            'ffmpeg: multiple formats support'
            'flac: FLAC transcoding support'
            'lame: MP3 transcoding support'
            'opus-tools: OPUS transcoding support'
            'sox: WAV transcoding support'
            'vorbis-tools: OGG transcoding support')
source=("${pkgname}-${pkgver}.tar.gz::https://github.com/Cygn/pulseaudio-dlna/archive/${pkgver}.tar.gz")
sha256sums=('e33ae82f77f58fdffa9df2c308cb649e386813d7eed7a013437c3cea1c024322')

prepare() {
  mv "pulseaudio-dlna-${pkgver}" "${pkgname}-${pkgver}"
  cd "${pkgname}-${pkgver}"
  sed -i '/dbus-python/d' setup.py
}

build() {
  cd "${pkgname}-${pkgver}"
  python setup.py build
}

package() {
  cd "${pkgname}-${pkgver}"
  install -D man/pulseaudio-dlna.1 -t "${pkgdir}/usr/share/man/man1/"
  python setup.py install --root="${pkgdir}" --optimize=1 --skip-build
}
```
and failed to run `pulseaudio-dlna` with these errors
```
/usr/lib/python3.10/site-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: 2.0.5-build-libtorrent-rasterbar-src-libtorrent-rasterbar-2.0.5-bindings-python is an invalid version and will not be supported in a future release
  warnings.warn(
Traceback (most recent call last):
  File "/usr/bin/pulseaudio-dlna", line 33, in <module>
    sys.exit(load_entry_point('pulseaudio-dlna==0.6.4', 'console_scripts', 'pulseaudio-dlna')())
  File "/usr/bin/pulseaudio-dlna", line 25, in importlib_load_entry_point
    return next(matches).load()
  File "/usr/lib/python3.10/importlib/metadata/__init__.py", line 162, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/lib/python3.10/site-packages/pulseaudio_dlna/__init__.py", line 19, in <module>
    import pkg_resources
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3266, in <module>
    def _initialize_master_working_set():
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3240, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 3278, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 573, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 891, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 777, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'casttube>=0.2.0' distribution was not found and is required by pychromecast
```
However `python-casttube 0.2.1` has been installed already, and 0.6.1 runs perfectly.

**Comment by Cygn** [comment]
I've bumped up to 0.6.5 which requires explicitly pychromecast > 10.  Note that pulseaudio dos not require directy casttube. 
Reminder that I don't know jack about pkgbuild - you are on your own for Arch related issues.
